apps/opencs/model/world/columnimp.cpp

```diff
@@ -89,10 +89,9 @@ namespace CSMWorld
 
         DataType values(Size, 0);
 
-        if (land.isDataLoaded(Land::DATA_WNAM))
+        for (int i = 0; i < Size; ++i)
         {
-            for (int i = 0; i < Size; ++i)
-                values[i] = land.mWnam[i];
+            values[i] = land.mWnam[i];
         }
```

Comment:
How about this:
```cpp
if (land.mDataTypes & Land::DATA_WNAM)
    std::copy(land.mWnam, land.mWnam + Size, values);
```

Reply:
std::copy fails with a curious error here, maybe because complex data type juggling through templates.

```
In file included from /usr/include/c++/9.1.0/bits/char_traits.h:39,
                 from /usr/include/c++/9.1.0/ios:40,
                 from /usr/include/c++/9.1.0/istream:38,
                 from /usr/include/c++/9.1.0/sstream:38,
                 from /home/nelsson/Koodi/openmw_code/transientlandediting/openmw/apps/opencs/model/world/columnimp.hpp:6,
                 from /home/nelsson/Koodi/openmw_code/transientlandediting/openmw/apps/opencs/model/world/columnimp.cpp:1:
/usr/include/c++/9.1.0/bits/stl_algobase.h: In instantiation of ‘_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = const signed char*; _OI = QVector<signed char>]’:
/usr/include/c++/9.1.0/bits/stl_algobase.h:441:30:   required from ‘_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = const signed char*; _OI = QVector<signed char>]’
/usr/include/c++/9.1.0/bits/stl_algobase.h:474:7:   required from ‘_OI std::copy(_II, _II, _OI) [with _II = const signed char*; _OI = QVector<signed char>]’
/home/nelsson/Koodi/openmw_code/transientlandediting/openmw/apps/opencs/model/world/columnimp.cpp:94:60:   required from here
/usr/include/c++/9.1.0/bits/stl_algobase.h:396:57: error: no type named ‘value_type’ in ‘struct std::iterator_traits<QVector<signed char> >’
  396 |       typedef typename iterator_traits<_OI>::value_type _ValueTypeO;
      |                                                         ^~~~~~~~~~~
/usr/include/c++/9.1.0/bits/stl_algobase.h:401:9: error: no type named ‘value_type’ in ‘struct std::iterator_traits<QVector<signed char> >’
  398 |       const bool __simple = (__is_trivially_copyable(_ValueTypeI)
      |                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  399 |         && __is_pointer<_II>::__value
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  400 |         && __is_pointer<_OI>::__value
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  401 |         && __are_same<_ValueTypeI, _ValueTypeO>::__value);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Still, I'd add the check for mDataTypes.

Reply:
bcb75de